core/src/main/java/org/elasticsearch/common/util/concurrent/EsThreadPoolExecutor.java

```diff
@@ -93,4 +98,31 @@ public void execute(Runnable command) {
             }
         }
     }
+
+    @Override
+    public String toString() {
+        /*
+         * ThreadPoolExecutor has some nice information in its toString but we
+         * can't recreate it without nastier hacks than this.
+         */
+        String tpeToString = super.toString();
+        int startOfInfoInTpeToString = tpeToString.indexOf('[');
+        String tpeInfo;
+        if (startOfInfoInTpeToString >= 0) {
+            tpeInfo = tpeToString.substring(startOfInfoInTpeToString + 1);
+        } else {
+            assert false: "Unsupported ThreadPoolExecutor toString";
+            tpeInfo = tpeToString;
+        }
+        StringBuilder b = new StringBuilder();
+        b.append(getClass().getSimpleName()).append('[');
+        b.append(name).append(", ");
+        if (getQueue() instanceof SizeBlockingQueue) {
+            @SuppressWarnings("rawtypes")
+            SizeBlockingQueue queue = (SizeBlockingQueue) getQueue();
+            b.append("queue capacity = ").append(queue.capacity()).append(", ");
+        }
+        b.append("state = ").append(tpeInfo);
+        return b.toString();
+    }
```

Comment:
Not really? Without something silly like this you can't have both the name of the name and the debugging information from the JDK's ThreadPoolExecutor. And the debugging information looks genuinely useful so I didn't want to break it by just returning the name.

The SizeBlockingQueue business is strait from EsAbortPolicy it just felt more right to combine it with the debugging information available in ThreadPoolExecutor's toString.